### PR TITLE
enable base prometheus metrics for template service broker

### PR DESCRIPTION
**Pull request by jim-minter**
https://trello.com/c/wDxqVOqy/1181-5-prometheus-metrics-for-template-broker-techdebt

### CODE REVIEW

File: examples/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -409,6 +409,22 @@ objects:
           action: replace
           target_label: kubernetes_name
 
+      - job_name: 'openshift-template-service-broker'
+
+        scheme: https
+        tls_config:
+          # TODO: specify CA
```
Comment (jim-minter) [inline examples/prometheus/prometheus.yaml]: @deads2k I think the TSB serving cert is signed by the service-signer CA - is it possible to get hold of that CA cert in the context of the prometheus pod?

File: pkg/templateservicebroker/openservicebroker/server/route.go [hunk 2]
```diff
@@ -48,6 +49,13 @@ func Route(container *restful.Container, path string, b api.Broker) {
 	ws.Route(ws.GET("/service_instances/{instance_id}/last_operation").To(shim(lastOperation)))
 	ws.Route(ws.PUT("/service_instances/{instance_id}/service_bindings/{binding_id}").To(shim(bind)))
 	ws.Route(ws.DELETE("/service_instances/{instance_id}/service_bindings/{binding_id}").To(shim(unbind)))
+
+	routes := ws.Routes()
+	for i, route := range routes {
+		// CHECK: ok to override resource with route.Path?
+		routes[i].Function = metrics.InstrumentRouteFunc(route.Method, route.Path, "", "cluster", route.Function)
```
Comment (jim-minter) [inline pkg/templateservicebroker/openservicebroker/server/route.go]: @deads2k I'd like to be able to instrument API calls against the TSB - do you think it's feasible to use "apiserver_request_count" for this?  Problem is that "apiserver_request_count" expects resource/subresource whereas we're dealing with slightly more arbitrary API paths - current worst case is something like "/service_instances/{instance_id}/service_bindings/{binding_id}"

File: examples/prometheus/prometheus.yaml (hunk 1 above)
Comment (deads2k) [reply examples/prometheus/prometheus.yaml]: > @deads2k I think the TSB serving cert is signed by the service-signer CA - is it possible to get hold of that CA cert in the context of the prometheus pod?

It's injected into the SA token secret.  It's the CA that isn't ca.crt.

File: pkg/templateservicebroker/openservicebroker/server/route.go (hunk 2 above)
Comment (deads2k) [reply pkg/templateservicebroker/openservicebroker/server/route.go]: It's a little weird, but since its your own servers, I suppose its fine.

**Comment by jim-minter** [comment]
I'm not wild about overloading resource with text like "/brokers/template.openshift.io/v2/service_instances/{instance_id}" (@bparees, @smarterclayton, thoughts?), but otherwise it's beneficial to be reusing the apiserver metrics routines and formats.

Example of the metrics this adds:

```
apiserver_request_count{client="curl/7.53.1",code="200",contentType="application/json",resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 1
apiserver_request_count{client="curl/7.53.1",code="200",contentType="application/json",resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 9
apiserver_request_count{client="curl/7.53.1",code="200",contentType="application/json",resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_count{client="curl/7.53.1",code="200",contentType="application/json",resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_count{client="curl/7.53.1",code="201",contentType="application/json",resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_count{client="curl/7.53.1",code="202",contentType="application/json",resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="125000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="250000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="2e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="4e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="500000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="8e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="+Inf"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="125000"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1e+06"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="250000"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="2e+06"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="4e+06"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="500000"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="8e+06"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="+Inf"} 9
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="125000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="1e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="250000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="2e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="4e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="500000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="8e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",le="+Inf"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="125000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="1e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="250000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="2e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="4e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="500000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="8e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",le="+Inf"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="125000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="1e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="250000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="2e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="4e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="500000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="8e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",le="+Inf"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="125000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="1e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="250000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="2e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="4e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="500000"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="8e+06"} 1
apiserver_request_latencies_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",le="+Inf"} 1
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 1
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 9
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_latencies_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 1
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 9
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE"} 1
apiserver_request_latencies_summary_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT"} 1
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",quantile="0.5"} 7785
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",quantile="0.9"} 7785
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",quantile="0.99"} 7785
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",quantile="0.5"} 4167
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",quantile="0.9"} 6832
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",quantile="0.99"} 6832
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.5"} 21390
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.9"} 21390
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.99"} 21390
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",quantile="0.5"} 24100
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",quantile="0.9"} 24100
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT",quantile="0.99"} 24100
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.5"} 5531
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.9"} 5531
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE",quantile="0.99"} 5531
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",quantile="0.5"} 21025
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",quantile="0.9"} 21025
apiserver_request_latencies_summary{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT",quantile="0.99"} 21025
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 7785
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 46859
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE"} 21390
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT"} 24100
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE"} 5531
apiserver_request_latencies_summary_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT"} 21025
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 7785
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 46859
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="DELETE"} 21390
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}",scope="cluster",subresource="",verb="PUT"} 24100
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="DELETE"} 5531
apiserver_request_latencies_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/service_bindings/{binding_id}",scope="cluster",subresource="",verb="PUT"} 21025
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1000"} 0
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="10000"} 0
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="100000"} 0
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1e+06"} 1
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1e+07"} 1
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1e+08"} 1
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="1e+09"} 1
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET",le="+Inf"} 1
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="100000"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="10000"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1000"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1e+06"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1e+07"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1e+08"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="1e+09"} 9
apiserver_response_sizes_bucket{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET",le="+Inf"} 9
apiserver_response_sizes_count{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 1
apiserver_response_sizes_count{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 9
apiserver_response_sizes_sum{resource="/brokers/template.openshift.io/v2/catalog",scope="cluster",subresource="",verb="GET"} 126285
apiserver_response_sizes_sum{resource="/brokers/template.openshift.io/v2/service_instances/{instance_id}/last_operation",scope="cluster",subresource="",verb="GET"} 259
```

**Comment by bparees** [comment]
it lgtm, i guess it's just an open question if @smarterclayton wants api metrics enabled for standalone api servers like the TSB.

**Comment by smarterclayton** [comment]
Let's hold on this, you're not an "api" per se and the api metrics aren't
necessarily appropriate for you.

**Comment by jim-minter** [comment]
Can I check, hold this pending what?  I believe that these metrics are relevant and valuable to the TSB component, and the work to add them is broadly done, so I'd like to understand what blocks this PR :)

**Comment by smarterclayton** [comment]
Broker is not an API, you aren't obeying the standard rules, so I don't want to spend time worrying about these metrics for 3.7.  TSB metrics are nice but not required.  Doing the base prometheus gatherers for process and http would be better.

**Comment by jim-minter** [comment]
updated:

```
http_request_duration_microseconds_count{handler="prometheus"} 9
http_request_duration_microseconds{handler="prometheus",quantile="0.5"} 1326
http_request_duration_microseconds{handler="prometheus",quantile="0.9"} 1640.69
http_request_duration_microseconds{handler="prometheus",quantile="0.99"} 1640.69
http_request_duration_microseconds_sum{handler="prometheus"} 12905.211999999998
http_request_size_bytes_count{handler="prometheus"} 9
http_request_size_bytes{handler="prometheus",quantile="0.5"} 62
http_request_size_bytes{handler="prometheus",quantile="0.9"} 62
http_request_size_bytes{handler="prometheus",quantile="0.99"} 62
http_request_size_bytes_sum{handler="prometheus"} 558
http_requests_total{code="200",handler="prometheus",method="get"} 9
http_response_size_bytes_count{handler="prometheus"} 9
http_response_size_bytes{handler="prometheus",quantile="0.5"} 29578
http_response_size_bytes{handler="prometheus",quantile="0.9"} 29584
http_response_size_bytes{handler="prometheus",quantile="0.99"} 29584
http_response_size_bytes_sum{handler="prometheus"} 266013
process_cpu_seconds_total 4.02
process_max_fds 1.048576e+06
process_open_fds 8
process_resident_memory_bytes 1.09973504e+08
process_start_time_seconds 1.506431001e+09
process_virtual_memory_bytes 8.23054336e+08
```

**Comment by smarterclayton** [comment]
Looks fine, we can iterate in the future.

**Pull request merged**